src/framework/utils.hpp

```diff
@@ -697,6 +707,105 @@ matrix<T> tensor_product(const matrix<T> &A, const matrix<T> &B) {
   return temp;
 }
 
+template <class T>
+matrix<T> concatenate (const matrix<T> &A, const matrix<T> &B, uint_t axis) {
+  // Works out a concatenate of two matrices A, B
+  // To add B as new rows axis == 0 (vertical)
+  // To add A as new cols axis == 1 (horizontal)
+  if (axis != 0 && axis!= 1) {
+    std::cerr << "MU::concatenate: axis must be 0 or 1" << std::endl;
+	exit(1);
```

Comment:
... and of course, we do not `exit` from the process.
Just remove this line.

Reply:
Done.